layers/+lang/scala/packages.el

```diff
@@ -191,6 +192,9 @@
       (when (configuration-layer/package-usedp 'expand-region)
         (require 'ensime-expand-region nil 'noerror)))))
 
+(defun scala/post-init-flycheck ()
+  (add-hook 'scala-mode-hook 'flycheck-mode))
```

Comment:
https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/python/packages.el#L97-L98

This is how you need to do it.